libraries/utils/utils/include/thread_pool.h

```diff
@@ -93,14 +96,14 @@ namespace bnb {
 
     // add new work item to the pool
     template<class F, class... Args>
-    auto thread_pool::enqueue(F&& f, Args&&... args) 
+    auto thread_pool::enqueue(F&& f, Args&&... args)
         -> std::future<typename std::result_of<F(Args...)>::type>
     {
         using return_type = typename std::result_of<F(Args...)>::type;
```

Comment:
-> std::future<typename std::invoke_result<F, Args...>::type>;

Reply:
yes missed that